models/items.js

```diff
@@ -0,0 +1,225 @@
+const firestore = require("../utils/firestore");
+
+const itemModel = firestore.collection("items");
+const tagModel = firestore.collection("tags");
+const levelModel = firestore.collection("levels");
+
+/**
+ *
+ * @param itemData { Object }: Data of the item, item type and Array of pairs of tagid and levelid
+ * @returns {Promise<tagModel|String>}
+ */
+
+const addTagsToItem = async (itemData) => {
+  try {
+    const { itemid, itemType, tagPayload } = itemData;
+    const batch = firestore.batch();
+    for (const tag of tagPayload) {
+      const tagData = await tagModel.doc(tag.tagid).get();
+      const levelData = await levelModel.doc(tag.levelid).get();
+      const itemTag = {
+        itemid,
+        itemtype: itemType.toUpperCase(),
+        tagid: tag.tagid,
+        tagname: tagData.data().name,
+        tagtype: tagData.data().type,
+        levelid: tag.levelid,
+        levelname: levelData.data().name,
+        levelnumber: levelData.data().levelnumber,
+      };
+      const docid = itemModel.doc();
+      batch.set(docid, itemTag);
+    }
+    await batch.commit();
+    return { itemid };
+  } catch (err) {
+    logger.error("Error in creating Item", err);
+    throw err;
+  }
+};
+
+/**
+ *
+ * @param itemData { Object }: itemid and tagid of the tag to be removed from the item
+ * @returns {Promise<tagModel|Object>}
+ */
+
+const removeTagsFromItem = async (itemData) => {
+  try {
+    const { itemid, tagid } = itemData;
+    const query = itemModel.where("tagid", "==", tagid).where("itemid", "==", itemid);
+    query.get().then(function (querySnapshot) {
+      querySnapshot.forEach(function (doc) {
+        doc.ref.delete();
+      });
+    });
+    return { itemid, tagid };
+  } catch (err) {
+    logger.error("Error in deleting Item", err);
+    throw err;
+  }
+};
+
+/**
+ *
+ * @param itemtype { String }: type of the item
+ * @returns {Promise<tagModel|Array>}
+ */
+
+const getItemsBasedOnType = async (itemtype) => {
+  try {
+    const data = await itemModel.where("itemtype", "==", itemtype.toUpperCase()).get();
```

Comment:
Same comment for selecting data as other PR#813.

Reply:
https://github.com/Real-Dev-Squad/website-backend/pull/815/commits/900ac7b5926375cbf6d894749f5caf922953c947 this is fixed here I guess